README.md

```diff
@@ -62,6 +62,13 @@ for the Object.
 
 _Note: The `publicPath` property is required, whereas all other options are optional_
 
+### acceptedMethods
```

Comment:
`options.methods` or even `options.methods.accept(ed)` to be consistent with naming `e.g options.headers` ? (Ideally we cold group it under a `options.http` prop but that's sadly not possible atm)

Reply:
> Ideally we cold group it under a options.http prop but that's sadly not possible atm
Is it a limitation caused by interaction with some other module? I ask that because it is possible, with an (otherwise) inconsequential change at https://github.com/webpack/webpack-dev-middleware/blob/ebdcedadbe485eb7cc4cfcacd57af544977cd2cb/lib/middleware.js#L27 
to make it
```javascript
const acceptedMethods = (context.options.http ? context.options.http.methods : null) || ['GET'];
```

The webpack.config would then be
```javascript
serve: {
  hotClient: { host:  { client: '*', server: '0.0.0.0' } },
  host: '0.0.0.0',
  devMiddleware: {
    publicPath: ProjectConfig.getOutput(target).publicPath,
    http: {
      methods: ['GET', 'POST']
    }
  },
  add: (app, middleware, options) => {
    const historyOptions = {};
    historyOptions.logger = console.log.bind(console);
    historyOptions.acceptedMethods = ['GET', 'POST'];
    app.use(convert(history(historyOptions)));
  }
},
```
I tested it in my environment and it works, what would be the impediment to use options.http.methods?

Reply:
🤔 I tink this could work out and would personally be in favor of making this change. It will likely involve some modifictaions to `webpack-dev-server` but that's ok from my side

cc @evilebottnawi What do you think about adding it to `options.http` instead. Anything of  concern with this approach ?

Reply:
It's definitely more organized and we could eventually move `options.headers` under `options.http`  (e.g `options.http.headers`) in the next major aswell then

Reply:
Yep, `options.methods` is equally fine aswell, we can eventually group them later in the next major, but it's a separate issue. So we all agree on moving forward with `options.methods` ? @ferdinando-ferreira ?